plugin/rpcplugin/api.go

```diff
@@ -26,6 +28,67 @@ func (h *LocalAPI) LoadPluginConfiguration(args struct{}, reply *[]byte) error {
 	return nil
 }
 
+type APITeamReply struct {
+	Team  *model.Team
+	Error *model.AppError
+}
+
+func (api *LocalAPI) GetTeamByName(args string, reply *APITeamReply) error {
+	team, err := api.api.GetTeamByName(args)
+	*reply = APITeamReply{
+		Team:  team,
+		Error: err,
+	}
```

Comment:
Why the structs and the setting of pointers instead of just returning the team and error?

Reply:
This is just how `net/rpc` works. RPC methods have to look like this:

```go
func (t *T) MethodName(argType T1, replyType *T2) error
```

See https://golang.org/pkg/net/rpc/